information-system/data-store/access-control.md

```diff
@@ -0,0 +1,101 @@
+
+### Who are general users?
+
+General Datastore users refers to the group of all users who can access the datastore. As such, general user access controls facilitate the general privacy or publicity (to all datastore users) of a dataset. You could limit access controls for general users if you wish to keep your dataset private to a group (perhaps during development processes and then later release it by updating general user access rights).
+
+### Where to manage general user access
+
+Upon registering a dataset, the general datastore user access rights can be managed by navigating to the 'settings' tab when viewing the dataset.
+
+In future versions of the IS, the ability to set access levels during the registration process will be implemented. For now, a workaround is to register a dataset with meaningless metadata, limit general access privileges, then meaningfully update the metadata as you wish.
+
+## Group access to dataset
+
+The group access feature enables custom permissions for groups (and thus members) to a dataset. This can achieve the privatisation of datasets to only certain users of the information system by *not* enabling general user read/write permissions and creating a group with enabled read/write permissions for its members.
+
+### Creating a group
+
+Group creation is currently performed by Information System Administrators. Please get in touch and we can help facilitate this for you.
+
+### Adding group access to datasets
+
+Groups access to datasets can be manually managed by navigating to the dataset's settings tab, then Access Control->Groups. Here, group permission can be added to the dataset. Group access levels for this dataset can also be edited here by selecting the group.
+
+Group access can be automatically applied with default values upon the creation of a new dataset. This occurs if 'Default data store access configuration' is enabled in the group creation process by the IS administrator. The result is that if a user belonging to that group creates a new dataset, then it will automatically be made available to the group according to the specified default access levels. Access levels can then be updated from default by navigating to the group in the dataset as mentioned in the above paragraph.
+
+A dataset can be made available to more than one group with differing accessibility levels (see note below). For example, you may wish to have a group of dataset contributors which will require write access, and another group for dataset viewing only, meanwhile the more general system users may not have access at all. 
+
+**Note**: If a user is accessing a dataset which has access rules for multiple groups of which the user is all apart of, the user will be given the more permissive access rights. This is because enabling an access right is seen as a "granting" action, meanwhile not enabling an access right is *not* seen as 'denying' access. Therefore, roles can only grant access, not deny.
```

Comment:
This can be made a bit clearer:

**Note**: When evaluating a user's access to a dataset, an optimistic approach is taken. For a given action, the following access checks are applied in this order:

1. Does the user have the correct data store access role for this action (see link to access roles doco)? If not, the **action is denied.** 
2. Does the user own this dataset? If so, the **action is approved**.
3. Does the "General Access" configuration for the dataset grant this action? If so, **action is approved**.
4. Does the user belong to **any** group which **grants** this action? If so, **action is approved**.
5. Otherwise, the **action is denied**.

Reply:
Are you suggesting replacing the whole **note** with these steps? 

I disagree that providing 5 steps clearer than the first sentence in the note. I don't think users need to know the order if it is functionally the same as them gaining access in line with their most permissive rights.

Reply:
Maybe you're right - I think my description above is better for our dev documentation. 

I think you're wording could be simplified in that case, maybe just

**Note** If a user is accessing a dataset which has access rules for multiple groups of which the user is a part of, the user will be granted the most permissive combination of those access rights.

Reply:
Went with above

Reply:
Resolved